### PR TITLE
Adding VsCode Vim to preview manifest.

### PR DESCRIPTION
**Pull request by adamskt**
Adding Vim goodness to SQL Ops Studio.

### CODE REVIEW

**Comment by MattIrv** [comment]
@llali @yualan did we ever figure out how we want to include existing VS Code extensions? This seems like a good candidate. Since this is on the VS Code marketplace already I wonder if it is possible for SQL Ops Studio to download directly from the marketplace instead of having to point users there to download it manually?

**Comment by llali** [comment]
@adamskt Thanks for your contribution!
At this moment, we don't have any plan for adding VS code extensions to our marketplace. Let me discuss this internally with our team and I'll get back to you as soon as possible.

**Comment by adamskt** [comment]
Your welcome!  I imagine you are in a bit of a tough spot-- some VS Code extensions are very applicable to the user experience (settings sync, Vim) and others are most definitely not (Typescript linter, Docker commands).  Either way, it was pretty easy to get some VS Code extensions loaded in my instance of SQL Ops Studio.

**Comment by llali** [comment]
@adamskt yes as you said it's easy to install any VS code extension manually using the "Install from VSIX" command. But to put them in our marketplace, we have to first make sure how to do that.

**Comment by adamskt** [comment]
Not a problem-- just want to make sure there was some discussion around it 😁

**Comment by llali** [comment]
@adamskt I'm planning to add this extension to our gallery. Can you please fix the conflict and use the latest release? Thanks

File: extensionsGallery.json [hunk 1]
```diff
@@ -407,6 +407,57 @@
...
+                    "lastUpdated": "05/16/2018",
+                    "assetUri": "",
+                    "fallbackAssetUri": "fallbackAssetUri",
+                    "files": [
+                    {
+                      "assetType": "Microsoft.SQLOps.DownloadPage",
+                      "source": "https://github.com/VSCodeVim/Vim/releases/tag/v0.12.0"
+                    },
+                    {
+                      "assetType": "Microsoft.VisualStudio.Services.Links.Source",
```
Comment (llali) [inline extensionsGallery.json]: please add this as a property bellow.
"properties": [
						  { "key": "Microsoft.VisualStudio.Code.ExtensionDependencies", "value":""},
						  { "key": "Microsoft.VisualStudio.Code.Engine", "value":"*"},
						  { "key": "Microsoft.VisualStudio.Services.Links.Source", "value":"https://github.com/VSCodeVim/Vim"}
						]

File: extensionsGallery.json [hunk 2]
```diff
@@ -407,6 +407,57 @@
...
+                    }
+                    ],
+                    "properties": [
+                      { "key": "Microsoft.VisualStudio.Code.ExtensionDependencies", "value":""},
+                      { "key": "Microsoft.VisualStudio.Code.Engine", "value":"*"}
+                    ]
+                  }
+                ],
+                "statistics": [],
                 "flags": "preview"
```
Comment (llali) [inline extensionsGallery.json]: this extension is not preview is it?

Comment (llali) [inline extensionsGallery.json]: if this extension is not preview please remove this flag

**Comment by adamskt** [comment]
Bugger... I freshened my fork from the wrong upstream branch...

File: extensionsGallery.json [hunk 3]
```diff
@@ -480,14 +480,66 @@
...
+							"lastUpdated": "07/11/2018",
+							"assetUri": "",
+							"fallbackAssetUri": "fallbackAssetUri",
+								"files": [
+								{
+									"assetType": "Microsoft.SQLOps.DownloadPage",
+									"source": "https://github.com/VSCodeVim/Vim/releases/tag/v0.15.0"
+								},
+								{
+									"assetType": "Microsoft.VisualStudio.Services.Links.Source",
```
Comment (llali) [inline extensionsGallery.json]: please remove this assetType. The correct property is added so we don't need this anymore

**Comment by progressiveae1** [comment]
I can't seem to find vsvim in the marketplace for azure data studio. Was this carried over after the rename?

**Comment by adriangonzalez** [comment]
@progressiveae1 image

**Comment by PiotrPerak** [comment]
I also can't find it. Was it removed?

**Comment by kburtram** [comment]
@PiotrPerak currently you need to install this extension using the "Install from VSIX" command.  The VSIX can be downloaded from https://marketplace.visualstudio.com/items?itemName=vscodevim.vim using the "Download Extension" option on the right sidebar.

It's a long story, but in summary we need to workout our entire extension marketplace story before including arbitrary VS Code extensions in the ADS marketplace.  In the interim please side-load any desired extensions.

**Comment by PiotrPerak** [comment]
@kburtram Unable to install version '1.3.0' of extension 'vscodevim.vim-1.3.0' as it is not compatible with Azure Data Studio '1.5.2'.

**Pull request merged**